src/store/socket_init.js

```diff
@@ -14,11 +14,28 @@ import {
 
 import {
   addChat,
-  addMessage,
+  receiveMessage,
 } from './Chat';
 
+let socket = null;
 
-export default function (store) {
+// Middleware for Redux to watch for new chat messages
```

Comment:
I'm looking forward to seeing how this works. Right now brain too tired for braining hard on things